src/Gir/Generation/IGeneratableExtensions.cs

```diff
@@ -124,20 +125,36 @@ public static (string both, string names) BuildParameters (this IMethodLike call
 				}
 
 				var symbol = parameter.Resolve(opts);
-				typeAndName.Add(symbol.CSharpType + " " + parameter.Name);
+				marshalTypeAndName.Add(symbol.CSharpType + " " + parameter.Name);
+				typeAndName.Add(symbol.CSharpType + (parameter.Array != null ? "[]" : "") + " " + parameter.Name);
 				parameterNames.Add(parameter.Name);
 			}
 
 			// PERF: Use an array as the string[] overload of Join is way more efficient than the IEnumerable<string> one.
+			string marshalParameterString = string.Join(", ", marshalTypeAndName.ToArray());
 			string parameterString = string.Join (", ", typeAndName.ToArray ());
 			string baseParams = string.Join (", ", parameterNames.ToArray ());
 
-			return (parameterString, baseParams);
+			return new BuildParametersResult(marshalParameterString, parameterString, baseParams);
 		}
 
 		static IEnumerable<IMemberGeneratable> GetMemberGeneratables (this IGeneratable gen)
 		{
 			return Utils.GetAllCollectionMembers<IMemberGeneratable> (gen);
 		}
 	}
+
+	public class BuildParametersResult
```

Comment:
Should we name this `ParametersResult` instead?

Reply:
👍